mfr/extensions/tabular/libs/stdlib_tools.py

```diff
@@ -30,3 +35,26 @@ def csv_stdlib(fp):
         })
     rows = [row for row in reader]
     return columns, rows
+
+
+def _set_dialect_quote_attrs(dialect, data):
+    """Set quote-related dialect attributes based on up to 2kb of csv data.
+
+    The regular expressions search for things that look like the beginning of
+    a list, wrapped in a quotation mark that is not dialect.quotechar, with
+    list items wrapped in dialect.quotechar and seperated by commas.
+
+    Example matches include:
+        "['1', '2', '3'         for quotechar == '
+        '{"a", "b", "c"         for quotechar == "
+    """
+    if dialect.quotechar == '"':
+        if re.search('\'[[({]\".+\",', data):
+            dialect.quotechar = "'"
+        if re.search('\'\'\'[[({]\".+\",', data):
```

Comment:
Fixed. 

Why using `r'...'` is a bad idea: [last paragraph](https://docs.python.org/2.0/ref/strings.html)

I could also check for the termination of the list, but I didn't because there's always a chance that it would exceed the 2kb read in to `data`.